tests/subsys/power/power_mgmt/src/main.c

```diff
@@ -0,0 +1,216 @@
+{
+	TC_PRINT("leave suspend\n");
+	zassert_true(notify_app_exit == true, "App doesn't be notified");
+	zassert_true(z_is_idle_thread_object(_current), NULL);
+	zassert_equal(state, SYS_POWER_STATE_SLEEP_1, NULL);
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+	struct device *dev;
+	uint32_t device_power_state;
+	/* at this point, devices are active again*/
+	dev = device_get_binding(DUMMY_DRIVER_NAME);
+	device_get_power_state(dev, &device_power_state);
+	zassert_equal(device_power_state, DEVICE_PM_ACTIVE_STATE, NULL);
+#endif
+	leave_idle = true;
+
+}
+
+/*
+ * @brief test power idle
+ *
+ * @details
+ *  - The global idle routine executes when no other work is available
+ *  - Provide a timeout parameter to the suspend routine indicating the
+ *    amount of time guaranteed to expire before the next timeout,
+ *    sys_pm_policy_next_state() handle this parameter.
+ *
+ * @see sys_pm_policy_next_state()
+ *
+ * @ingroup power_tests
+ */
+void test_power_idle(void)
+{
+	TC_PRINT("give way to idle thread\n");
+	k_sleep(SLEEP_TIMEOUT);
+}
+
+/*
+ * @brief test power state transition
+ *
+ * @details
+ *  - The system support control of power state ordering between
+ *    subsystems and devices
+ *  - The application can control system power state transitions in idle thread
+ *    through sys_pm_notify_power_state_entry and sys_pm_notify_power_state_exit
+ *
+ * @see sys_pm_notify_power_state_entry(), sys_pm_notify_power_state_exit()
+ *
+ * @ingroup power_tests
+ */
+void test_power_state_trans(void)
+{
+	enter_low_power = true;
+	TC_PRINT("give way to idle thread\n");
+	k_sleep(SLEEP_TIMEOUT);
+	zassert_true(leave_idle, NULL);
+}
+
+/*
+ * @brief notification between system and device
+ *
+ * @details
+ *  - device driver notify its power state change by device_pm_get and
+ *    device_pm_put
+ *  - system inform device system power state change through device interface
+ *    device_pm_control
+ *
+ * @see device_pm_get(), device_pm_put(), device_set_power_state(),
+ *      device_get_power_state()
+ *
+ * @ingroup power_tests
+ */
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+void test_power_state_notification(void)
+{
+	struct device *dev;
+	struct dummy_driver_api *api;
+	int ret;
+	uint32_t device_power_state;
+
+	dev = device_get_binding(DUMMY_DRIVER_NAME);
+	device_get_power_state(dev, &device_power_state);
+	zassert_equal(device_power_state, DEVICE_PM_ACTIVE_STATE, NULL);
+
+	api = (struct dummy_driver_api *)dev->driver_api;
+	ret = api->open(dev);
+	if (ret == 0)
+		api->close(dev);
+	device_get_power_state(dev, &device_power_state);
+	zassert_equal(device_power_state, DEVICE_PM_SUSPEND_STATE, NULL);
+}
+#else
+void test_power_state_notification(void)
+{
+	ztest_test_skip();
+}
+#endif
+
+/*
+ * @brief Power Management
+ *
+ * @details
+ *   - The system provide control over and notification of changes to
+ *     system power state
+ *
+ * @ingroup all_test
+ */
+void test_main(void)
+{
+	ztest_test_suite(power_test,
```

Comment:
This name could be misleading. This suite does not test power consumption as the name may suggest, but rather the power management subsystem. What do others think?

Reply:
not gating issue